pkgs/by-name/ri/ringracers/package.nix

```
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, fetchzip
+, fetchFromGitHub
+, cmake
+, curl
+, nasm
+, game-music-emu
+, libpng
+, SDL2
+, SDL2_mixer
+, libvpx
+, libyuv
+, zlib
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ringracers";
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner = "KartKrewDev";
+    repo = "RingRacers";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-fiKZg/WU8PRyR8RggYHSL0rVUBeFuuQdEKyIj/O0aN4=";
+  };
+
+  assets = stdenv.mkDerivation {
+    pname = "ringracers-data";
+    version = finalAttrs.version;
+
+    src = fetchzip {
+      url = "https://github.com/KartKrewDev/RingRacers/releases/download/v${finalAttrs.version}/Dr.Robotnik.s-Ring-Racers-v${finalAttrs.version}-Assets.zip";
+      hash = "sha256-qTWDp5jfpBtEcYmzsxeE6XaohjCjSl/x8eXxnIdhCxY=";
+      stripRoot = false;
+    };
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share/ringracers
+      cp -r * $out/share/ringracers
+
+      runHook postInstall
+    '';
+  };
```

If there's a no special reason to use the `/share/ringracers` structure, then you could just use `fetchzip` by itself.
`fetchzip` already outputs a derivation that is acceptable for usage as an `assets` derivation. Though if you do this, you'll need to remove the `/share/ringracers` part from the `makeWrapper` section.

```suggestion
  assets = fetchzip {
    name = "${finalAttrs.pname}-${finalAttrs.version}-assets";
    url = "https://github.com/KartKrewDev/RingRacers/releases/download/v${finalAttrs.version}/Dr.Robotnik.s-Ring-Racers-v${finalAttrs.version}-Assets.zip";
    hash = "sha256-qTWDp5jfpBtEcYmzsxeE6XaohjCjSl/x8eXxnIdhCxY=";
    stripRoot = false;
  };
```

---

Good catch